compiler/src/formatting/format.re

```diff
@@ -28,6 +28,33 @@ type expression_parent_type =
 
 let exception_primitives = [|"throw", "fail", "assert"|];
 
+let is_maths_op = fn =>
+  switch (fn) {
+  | "*"
+  | "/"
+  | "%"
+  | "+"
+  | "-"
+  | "<"
+  | ">"
+  | "&"
+  | "^"
+  | "|" => true
+  | _ => false
+  };
+
+let is_logic_op = fn =>
+  switch (fn) {
+  | "<="
+  | ">="
+  | "=="
+  | "!="
+  | "is"
+  | "&&"
+  | "||" => true
+  | _ => false
+  };
```

Comment:
These two functions need to be extended to the start of the operators in the same way that `op_precedence` does, but otherwise this looks good.

Reply:
That's done.